libs/forms/core/src/form-tree-node.service.ts

```diff
@@ -92,46 +162,71 @@ export class DynFormTreeNode<TControl extends AbstractControl = FormGroup>{
     // register the name to build the form path
     this._name = config.name ?? '';
 
+    // store the matchers to be processed afterViewInit
+    this._matchers = config.options?.matchers;
+
     if (!this.isolated) {
       // register the node with its parent
       this.parent?.addChild(this);
     }
   }
 
-  unregister(): void {
+  afterViewInit(): void {
+    // process the stored matchers
+    this._matchers?.map((config) => {
+      const matcher = this.formMatchers.getMatcher(config.matcher);
+
+      combineLatest(
+        // build an array of observables to listen changes into
+        config.when
+          .map(condition => this.formMatchers.getCondition(condition)) // handler fn
+          .map(fn => fn(this)) // condition observables
+      ).pipe(
+        map(results => config.operator === 'OR'
+          ? results.some(Boolean)
+          : results.every(Boolean)
+        ),
+        takeUntil(this._unsubscribe),
+        distinctUntilChanged(),
+      )
+      .subscribe(hasMatch => {
+        // run the matcher with the conditions result
+        matcher(this, config.negate ? !hasMatch : hasMatch)
+      });
```

Comment:
This is another approach to [ng-dynamic-forms](https://github.com/udos86/ng-dynamic-forms/blob/8ff9c596cb884196a6ba51245a630b0af6d2638e/projects/ng-dynamic-forms/core/src/lib/service/dynamic-form-relation.service.ts#L98-L124)